File: game-rest-api/src/it/klotski/web/game/configs/Board.java
package it.klotski.web.game.configs;

import it.klotski.web.game.domain.tile.ITile;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.RequiredArgsConstructor;
import lombok.With;

import java.util.TreeSet;

/**
 * Rappresenta la griglia di gioco.
 */
@With
@Getter
@RequiredArgsConstructor
@EqualsAndHashCode
public class Board {
    /**
     * La coordinata x in alto a sinistra del pezzo "winning tile" che determina la vittoria della partita.
     */
    private final int winningX;

    /**
     * La coordinata y in alto a sinistra del pezzo "winning tile" che determina la vittoria della partita.
     */
    private final int winningY;

    /**
     * Larghezza del tabellone.
     */
    private final int boardWidth;

    /**
     * Altezza del tabellone.
     */
    private final int boardHeight;

    /**
     * ID del tabellone.
     */
    private final int id;

    /**
     * Insieme ordinato di tessere (tiles) presenti nel tabellone.
     */
    private final TreeSet<ITile> tiles;
}

File: web-client/src/types/models/Board.ts
import Tile from "@/types/models/Tile.ts";

/**
 * @class Board
 * @classdesc Rappresenta una scacchiera di gioco.
 */
export default class Board {
    /**
     * @member {number} boardWidth - Larghezza della scacchiera.
     */
    boardWidth: number;

    /**
     * @member {number} boardHeight - Altezza della scacchiera.
     */
    boardHeight: number;

    /**
     * @member {number} id - ID della scacchiera.
     */
    id: number;

    /**
     * @member {Tile[]} tiles - Array di piastrelle sulla scacchiera.
     */
    tiles: Tile[];

    /**
     * @constructor
     * @param {number} boardWidth - Larghezza della scacchiera.
     * @param {number} boardHeight - Altezza della scacchiera.
     * @param {number} id - ID della scacchiera.
     * @param {Tile[]} tiles - Array di piastrelle sulla scacchiera.
     */
    constructor(boardWidth: number, boardHeight: number, id: number, tiles: Tile[]) {
        this.boardWidth = boardWidth;
        this.boardHeight = boardHeight;
        this.id = id;
        this.tiles = tiles;
    }

}